### PR TITLE
IR-195: add topology spread constraint to spec

### PR DESCRIPTION
**Pull request by flavianmissi**
relates to https://issues.redhat.com/browse/IR-195.

### CODE REVIEW

File: imageregistry/v1/zz_generated.deepcopy.go
Comment (flavianmissi) [inline imageregistry/v1/zz_generated.deepcopy.go:1]: this is go 1.17+ new build constraint syntax btw, see https://pkg.go.dev/cmd/go#hdr-Build_constraints

Comment (flavianmissi) [reply imageregistry/v1/zz_generated.deepcopy.go]: looks like CI didn't like it - removed.

File: imageregistry/v1/types.go
Comment (dmage) [inline imageregistry/v1/types.go:110]: should it be `spread`?

Comment (flavianmissi) [reply imageregistry/v1/types.go:110]: oh yeah, good catch!

Comment (flavianmissi) [reply imageregistry/v1/types.go]: fixed.

**Comment by dmage** [comment]
/assign @bparees 
For API review and approvals.

/lgtml

/hold
As the image registry team follows no-FF approach, we can merge PRs only once they are approved by QE, Docs, and Px. In this case we need to get approvals on the PR to openshift/cluster-image-registry-operator before we can merge this PR. Until then, you can use `replace` in your PRs in go.mod to use this API changes. Removing this `replace` will be the last step before we merge cluster-image-registry-operator's PR.

**Comment by dmage** [comment]
/retitle IR-195: add topology spread constraint to spec

**Comment by bparees** [comment]
/unassign

i believe all apireviews are presently going through @deads2k or @sttts

File: imageregistry/v1/00-crd.yaml
Comment (xiuwang) [inline imageregistry/v1/00-crd.yaml:866]: s/It''s /It's

File: imageregistry/v1/00-crd.yaml
Comment (xiuwang) [inline imageregistry/v1/00-crd.yaml:873]: `doesn''t , won''t , it''s`
Hi , if it's a common use here?

File: imageregistry/v1/00-crd.yaml
Comment (flavianmissi) [reply imageregistry/v1/00-crd.yaml:866]: The double quote (`''`) here is used to escape the single quote character (`'`), since the whole string is wrapped with single quotes.
See https://yaml.org/spec/1.2.2/#rule-c-quoted-quote

File: imageregistry/v1/00-crd.yaml
Comment (flavianmissi) [reply imageregistry/v1/00-crd.yaml:873]: Same as above

File: imageregistry/v1/00-crd.yaml
Comment (flavianmissi) [reply imageregistry/v1/00-crd.yaml:866]: I was confused about this myself - there's a better way to do this, which is to wrap the string with double quotes (`"`) instead, but since this file is auto-generated I don't think it's worth it because it'll be overwritten next time the schema changes.

File: authorization/v1/generated.pb.go
Comment (flavianmissi) [inline authorization/v1/generated.pb.go:1170]: Not sure why I got this change... ran `make update-codegen-crds` with go 1.17.2.

Comment (flavianmissi) [reply authorization/v1/generated.pb.go]: rolled back since the verify job broke

**Review by soltysh — APPROVED** [review]
/approve

**Comment by KeenonLee** [comment]
The TEST of SNO Scenario will continue tomorrow

**Pull request merged**
